hugo/content/posts/nix-bazel-crosspiling.md

```diff
@@ -0,0 +1,129 @@
+2. Fill in URL `https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/`
+   `{Sha1Sum}/debian_{version}_arm64_sysroot.tar.xz`
+
+**universal-apple-darwin**
+
+Pulled from Xcode installation on a Mac.
+
+1. Run `(export OUTPUT=${PWD}/sdks/universal-apple-darwin/{version}.tar.xz`
+   `&& cd $(xcode-select -p)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX{version}.sdk`
+   `&& tar -cvJf ${OUTPUT} .)`
+
+**universal-apple-ios**
+
+Pulled from Xcode installation on a Mac.
+
+1. Run `(export OUTPUT=${PWD}/sdks/universal-apple-ios/{version}.tar.xz`
+   `&& cd $(xcode-select -p)/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS{version}.sdk`
+   `&& tar -cvJf ${OUTPUT} .)`
+
+**universal-linux-android**
+
+Android NDK from https://developer.android.com/ndk/downloads
+
+**wasm32-unknown-unknown**
+
+By definition this shouldn't really require an SDK, but it does require `libclang_rt.builtins-*` which exist as a prebuilt in the `wasm32-wasi` SDK.
+https://github.com/WebAssembly/wasi-sdk/releases
+
+**wasm32-wasi**
+
+Download SDK and `libclang_rt.builtins-*`.
+https://github.com/WebAssembly/wasi-sdk/releases
+
+**x86_64-pc-windows-msvc**
+
+Generated via `xwin` tool. See https://jake-shadle.github.io/xwin/
+
+1. Install `xwin` by running `cargo install --locked xwin`
+2. Run `(export OUTPUT=${PWD}/sdks/x86_64-pc-windows-msvc/{version}.tar.xz`
+   `&& rm -rf /tmp/xwin`
+   `&& mkdir /tmp/xwin`
+   `&& xwin --accept-license --manifest-version {version} --temp splat --output /tmp/xwin`
+   `&& cd /tmp/xwin && tar -cvJf ${OUTPUT} .`
+   `&& rm -rf /tmp/xwin)`
+
+**x86_64-unknown-linux-gnu**
+
+Downloaded from Chromium project. Reverse engineered from https://chromium.googlesource.com/chromium/src/build/+/refs/heads/main/linux/sysroot_scripts/install-sysroot.py
+
+1. Get `Sha1Sum` for `debian_{version}_amd64_sysroot.tar` from https://chromium.googlesource.com/chromium/src/build/+/refs/heads/main/linux/sysroot_scripts/sysroots.json
+2. Fill in URL `https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/`
+   `{Sha1Sum}/debian_{version}_amd64_sysroot.tar.xz`
+
+**x86_64-unknown-nixos-gnu**
+
+Directly assembled from Nix packages inside of `flake.nix`.
+
+# Bazel
+
+Bazel uses all of the tools and SDKs exposed by Nix to build C/C++ and Rust code. All the Nix derivations under the `bazel` set are imported into the Bazel Workspace with [nixpkgs_flake_package](https://github.com/tweag/rules_nixpkgs#nixpkgs_flake_package). From there, a collection of Bazel toolchains are configured and registered.
+
+Unfortunately I couldn't figure out a way to configure C/C++ toolchains using Bazel's existing helper functions that I was happy with, so I ended up avoiding the Bazel "feature" system and configuring actions from scratch. While this lacks the dynamic configurability in the default Bazel C/C++ toolchains, I think it's a lot easier to follow and philosophically I try to keep as much out of Bazel CLI/`.bazelrc` flags as possible.
```

Comment:
Why do you try to keep as much out of those flags as possible? Do you care to elaborate, or would it be another blog post?